src/transport/subscription-worker/components/subscribe-request.ts

```diff
@@ -117,22 +117,25 @@ export class SubscribeRequest extends BasePubNubRequest {
    * @param timetokenOverride - Timetoken which should be used to patch timetoken in initial response.
    * @param timetokenRegionOverride - Timetoken origin which should be used to patch timetoken origin in initial
    * response.
+   * @param [cachedState] - Previously cached user's presence state for channels and groups.
    * @returns Aggregated subscribe request which will be sent.
    */
   static fromRequests(
     requests: SubscribeRequest[],
     accessToken?: AccessToken,
     timetokenOverride?: string,
     timetokenRegionOverride?: string,
+    cachedState?: Record<string, Payload>,
   ) {
     const baseRequest = requests[Math.floor(Math.random() * requests.length)];
+    const isInitialSubscribe = (baseRequest.request.queryParameters!.tt ?? '0') === '0';
+    const state: Record<string, Payload> = isInitialSubscribe ? (cachedState ?? {}) : {};
```

Comment:
So it preserves previous state even when subscription reset happens. I wonder should it be preserved.!

Reply:
@mohitpubnub it preserves explicitly set presence `state` which later will be used to override state received in subscribe requests and will use it only on initial subscription.